lib/types.ts

```diff
@@ -93,15 +93,49 @@ export type DeepWritable<T> = T extends Primitive
 interface WritableSet<ItemType> extends Set<DeepWritable<ItemType>> {}
 interface WritableMap<KeyType, ValueType> extends Map<DeepWritable<KeyType>, DeepReadonly<ValueType>> {}
 
-/**
- * Omit given key in object type
- * @deprecated Starting with TypeScript 3.5, Omit is natively available.
- */
-export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+/** Similar to the builtin Omit, but checks the filter strictly. */
+export type StrictOmit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
 
 /** Omit all properties of given type in object type */
 export type OmitProperties<T, P> = Pick<T, { [K in keyof T]: T[K] extends P ? never : K }[keyof T]>;
 
+/** Recursively omit deep properties */
+export type DeepOmit<T extends DeepOmitModify<P>, P> = T extends Primitive | Function | Date
```

Comment:
`T` and `K` are ok, but could you use names for the other generic params? Like `Filter` instead of `P`, `Map<infer KeyT, infer ValT>`, `Array<infer ElemT>`, etc. Makes the code easier to read, less like a complex math formula :)

Reply:
The usage of Map<K, V> and Set<U> is consistent with other Deep-* types.

But I think I should rename P to Filter. This change can help a lot.

Reply:
@quezak I've made a new commit, please see if it is better now.